pkg/resources/external_oauth_integration.go

```diff
@@ -41,548 +46,777 @@ var oauthExternalIntegrationSchema = map[string]*schema.Schema{
 	if err != nil {
-		return fmt.Errorf("error executing create statement: %w", err)
+		return nil, err
 	}
 
-	d.SetId(ExternalOauthIntegrationID(&input.ExternalOauthIntegration3))
-
-	return ReadExternalOauthIntegration(d, meta)
-}
-
-// ReadExternalOauthIntegration implements schema.ReadFunc.
-func ReadExternalOauthIntegration(d *schema.ResourceData, meta interface{}) error {
-	manager, err := snowflake.NewExternalOauthIntegration3Manager()
+	integrationProperties, err := client.SecurityIntegrations.Describe(ctx, id)
 	if err != nil {
-		return fmt.Errorf("couldn't create external oauth integration builder: %w", err)
+		return nil, err
 	}
 
-	input := ExternalOauthIntegrationIdentifier(d.Id())
-
-	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
+	if err = d.Set("name", integration.Name); err != nil {
+		return nil, err
+	}
+	if err = d.Set("enabled", integration.Enabled); err != nil {
+		return nil, err
+	}
+	if oauthType, err := integration.SubType(); err == nil {
+		if err = d.Set("external_oauth_type", oauthType); err != nil {
+			return nil, err
+		}
+	}
+	if prop, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "EXTERNAL_OAUTH_ISSUER" }); err == nil {
+		if err = d.Set("external_oauth_issuer", prop.Value); err != nil {
+			return nil, err
+		}
+	}
+	if prop, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "EXTERNAL_OAUTH_JWS_KEYS_URL"
+	}); err == nil {
+		if err = d.Set("external_oauth_jws_keys_url", listValueToSlice(prop.Value, false)); err != nil {
+			return nil, err
+		}
+	}
+	if prop, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "EXTERNAL_OAUTH_ANY_ROLE_MODE"
+	}); err == nil {
+		if err = d.Set("external_oauth_any_role_mode", prop.Value); err != nil {
+			return nil, err
+		}
+	}
+	if prop, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "EXTERNAL_OAUTH_RSA_PUBLIC_KEY"
+	}); err == nil {
+		if err = d.Set("external_oauth_rsa_public_key", prop.Value); err != nil {
+			return nil, err
+		}
+	}
+	if prop, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "EXTERNAL_OAUTH_RSA_PUBLIC_KEY_2"
+	}); err == nil {
+		if err = d.Set("external_oauth_rsa_public_key_2", prop.Value); err != nil {
+			return nil, err
+		}
+	}
 
-	// This resource needs a SHOW and a DESCRIBE
+	if prop, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "EXTERNAL_OAUTH_BLOCKED_ROLES_LIST"
+	}); err == nil {
+		// handle blocked_roles_list
+		params, err := client.Parameters.ShowParameters(ctx, &sdk.ShowParametersOptions{
```

Comment:
There a function in client.Parameters to call for one Parameter, use that one.

Reply:
Done.